internal/internal_schedule_client.go

```diff
@@ -719,9 +719,37 @@ func convertFromPBScheduleCalendarSpecList(calendarSpecPB []*schedulepb.Structur
 	return calendarSpec
 }
 
+func applyScheduleCalendarSpecDefault(calendarSpec *ScheduleCalendarSpec) {
+	if calendarSpec.Second == nil {
```

Comment:
I guess it's okay that a non-nil but zero-length slice doesn't get defaulted here?

Reply:
No, we should fix this IMO. Any time we're defaulting a slice we should use a len == 0 check.

Reply:
That was an intentional choice. If someone set an empty array they explicitly set the value and we should not override it. This also aligns with out other SDKs

Reply:
Hrmm, this is not usually a good Go practice for field defaults. I've found expecting a user to differentiate between nil and empty because they have different behaviors is rough. We can keep it, but I wouldn't be surprised if you'll have to explain it away (e.g. "I load this from a config file and the unmarshaller sets this to empty array"). Other langs do not have a precedent of treating empty and unset the same like Go and Proto and some others do.

Reply:
An empty slice and a nil slice are different things and it isn't unusual for APIs to treat them differently. `json.Marshal` treats them differently for example. The same confusion could also happen the other way if someone wants to specify and empty list (or a higher level API wrapping `ScheduleCalendarSpec` does) and we override it.